File: 2249-count-lattice-points-inside-a-circle/2249-count-lattice-points-inside-a-circle.rs
use std::collections::HashSet;

impl Solution {
    pub fn count_lattice_points(circles: Vec<Vec<i32>>) -> i32 {
        let mut set = HashSet::new();
        
        for circle in circles{
            let x = circle[0];
            let y = circle[1];
            let r = circle[2];
            
            for px in (x - r)..=(x + r){
                for py in (y - r)..=(y+ r){
                    
					// difference from the center of the circle
                    let delta_x = x - px;
                    let delta_y = y - py;
                    
					// using pythagoras theorem ( x^2 + y^2 = z^2 )
					// to get the point's (px, py) magnitude from the center of the circle
					
                    let magnitude = ((delta_x.pow(2) + delta_y.pow(2)) as f64).sqrt();
                    
                    if(magnitude <= r as f64){
                        set.insert((px, py));
                    }
                }
            }
        }
        
        set.len() as i32
    }
}

File: 2249-count-lattice-points-inside-a-circle/2249-count-lattice-points-inside-a-circle.py
class Solution:
    def countLatticePoints(self, circles: List[List[int]]) -> int:
        s= set()
        
        for (x, y, r) in circles:
            for i in range(x - r, x + r + 1):
                for j in range(y - r, (y + r) + 1):
                    dx = abs(i - x)
                    dy = abs(j - y)
                    
                    if sqrt(pow(dx, 2) + pow(dy, 2)) <= r:
                        s.add((i, j))
        # print(s)
        return len(s)